### ISSUE TITLE
[BUG] Missing http proxy settings on rke2 and rancher pod

### ISSUE THREAD

**Issue by bashofmann** [harvester/harvester#1218]
**Describe the bug**
When you run Harvester in an airgapped environment, where you only have internet connectivity through an http proxy, the proxy settings that are set during the installation process are not applied to RKE2 or the Rancher deployment.

**To Reproduce**
Steps to reproduce the behaviour:
* Install Harvester in an airgapped environment
* Provide the necessary http_proxy, https_proxy and no_proxy settings
* Check if the Rancher Deployment/Pods have these environment variables configured
* Check if the rke2-server systemd unit has these environment variables configured

**Expected behavior**
The environment variables should be there.

**Environment:**
 - Harvester ISO version: 0.3.0-preview1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): baremetal

**Additional context**
If they are missing, RKE2 can not pull/update additional container images from the internet. Also the remote UI functionality of Rancher, plus additional features where Rancher connects to the internet (KDM, chart repos, fleet git repos, ...) do not work.

For 0.3.0-preview1 this means that rancher answers with A HTTP 200 response code and a blank body, because it can not fetch the remote index.html.

From the UI side, it would be great to have an HTTP 5xx response code with a sensible error message and something in visible in the error logs.

**Comment by gitlawr** [comment]
~~Depending on https://github.com/rancher/rke2/issues/1834~~

**Comment by yasker** [comment]
~~@gitlawr What's the impact on missing the RKE2 proxy part? We cannot start any Pods, except for the internal packaged images? Does RKE1 work with proxy?~~

~~Also, is there any manual workaround for this problem?~~

The above questions might not be necessary anymore since it seems RKE2 team has fixed the problem. Now we need the reproduce steps to validate.

**Comment by gitlawr** [comment]
Also related to https://github.com/rancher/rancher/issues/34908

**Comment by guangbochen** [comment]
move this one to v1.0.0 since we need to wait for the https://github.com/rancher/rancher/issues/34908 to fully address the air-gap env support of the Rancher RKE2 server.

**Comment by janeczku** [comment]
**Workaround for container image downloading via a HTTP proxy**

In the Harvester install config YAML add the following write_files stanza (see [docs](https://docs.harvesterhci.io/v0.3/install/harvester-configuration/#oswrite_files)):

```
os:
  write_files:
  - path: /etc/default/rke2-server
    content: |
      # Environment variables
      HTTP_PROXY=http://proxy-host:port
      HTTPS_PROXY=http://proxy-host:port
      NO_PROXY=localhost,127.0.0.1,0.0.0.0,10.0.0.0/8,192.168.0.0/16,cattle-system.svc,.svc,.cluster.local,<internal domain>
    owner: root:root
    permissions: '0644'
   - path: /etc/default/rke2-agent
    content: |
      # Environment variables
      HTTP_PROXY=http://proxy-host:port
      HTTPS_PROXY=http://proxy-host:port
      NO_PROXY=localhost,127.0.0.1,0.0.0.0,10.0.0.0/8,192.168.0.0/16,cattle-system.svc,.svc,.cluster.local,<internal domain>
    owner: root:root
    permissions: '0644'
```

**Comment by TachunLin** [comment]
From `master-119c2c48-head` version, harvester provide a feature to let user configure http-proxy in the settings page.
To display this setting, harvester need to be installed with `internet` connection 
![image](https://user-images.githubusercontent.com/29251855/141323879-abad5c08-cfd0-4f88-af28-5ce7e8769b40.png)

However I found if harvester in installed in `airgapped` environment only have internet connectivity through an http proxy, we can't see the `http-proxy` configuration in settings page
![image](https://user-images.githubusercontent.com/29251855/141324017-b2af2261-a993-41d8-9e9b-4272a5ed5e43.png)

That is to say when we run Harvester in an `airgapped` environment, we can't set harvester to apply `http-proxy` in settings page.

I would suggest we can further check this setting to fit user scenario, please correct me if I misunderstood anything.

**Comment by WuJun2016** [comment]
You can wait for the backend version to be updated before testing the airgapped environment.

Maybe the front-end resources inside the latest back-end iso are not the latest. Wait for the back-end merge new pr and then test

**Comment by gitlawr** [comment]
To test:
1. Install Harvester in a private network that can access internet via an HTTP proxy
2. configure HTTP proxy in Harvester settings
3. Verify downloading VM images works.
4. Verify VM backup to external S3 works
5. Run `kubectl create deployment nginx --image=nginx:latest` on Harvester cluster and verify the image can be pulled and the nginx pod runs successfully

Note:
We don't propagate the proxy settings to the embedded rancher pods given that it is not something exposed to and affect users.

**Comment by TachunLin** [comment]
Verified the issue `partial fixed` on harvester master-f5bde7fa-head since it did not pass all the scenarios 
Move to implement pipeline for the check.

### Result
The new http proxy harvester setting, pass the following scenario

- [x]  Create image from remote URL
![image](https://user-images.githubusercontent.com/29251855/142584644-c7321df5-798d-4a76-843f-33810bd2d759.png)

- [x]  Pull image from internet and deploy nginx pod in running status

```
harvester-node-0:/home/rancher # kubectl create deployment nginx --image=nginx:latest
deployment.apps/nginx created
harvester-node-0:/home/rancher # kubectl get pods 
NAME                     READY   STATUS    RESTARTS   AGE
nginx-55649fd747-hkfjh   1/1     Running   0          28s
```

But can't pass the backup to S3 scenario
- [ ]  Backup to external S3 storage 

Already tested the version master-f5bde7fa-head can backup to external S3 with online harvester cluster. 
While in airgapped with http proxy, the backup will always keep in `progressing` 

![image](https://user-images.githubusercontent.com/29251855/142723714-cc6b6619-9e71-472a-81d9-a0cbfb17e14a.png)


### Test Information
* Test Environment: 3 nodes harvester on provo bare machine
* Harvester version: master-f5bde7fa-head


### Verify Steps

1. Clone the ipxe example in the offline feature branch
2. Create 3 nodes harvester by using vagrant pxe example
3. Open settings
4. Provide the following value in the http-proxy settings
```
{
  "httpProxy": "http://192.168.0.254:3128",
  "httpsProxy": "http://192.168.0.254:3128",
  "noProxy": "localhost,127.0.0.1,0.0.0.0,10.0.0.0/8,cattle-system.svc,192.168.0.0/16,.svc,.cluster.local,example.com"
}
```
5. Create image from URL 
https://cloud-images.ubuntu.com/focal/20211116/focal-server-cloudimg-amd64.img
6. ssh to harvester server node
7. Run `kubectl create deployment nginx --image=nginx:latest` on Harvester cluster
8. Run `kubectl get pods` 
9. Create a virtual machine
10. Set S3 back target setting
```
{
  "type": "s3",
  "endpoint": "",
  "accessKeyId": "",
  "secretAccessKey": "",
  "bucketName": "davidtclin-test",
  "bucketRegion": "us-east-1",
  "cert": "",
  "virtualHostedStyle": false
}
```
11. Take backup from virtual machine  

### Additional Context
Harvester node logs
[node01.log](https://github.com/harvester/harvester/files/7576100/node01.log)

**Comment by TachunLin** [comment]
Verified the issue `fixed` on harvester master-1e042e04-head (12/01) . It pass all the test scenarios. Close this issue 

### Result

- [x]  Create image from remote URL
![image](https://user-images.githubusercontent.com/29251855/144204032-ff117b28-4e1a-4c45-ac91-2df8a4ed3719.png)

- [x]  Pull image from internet and deploy nginx pod in running status

```
harvester-node-0:/home/rancher # kubectl create deployment nginx --image=nginx:latest
deployment.apps/nginx created
harvester-node-0:/home/rancher # kubectl get pods
NAME                              READY   STATUS    RESTARTS   AGE
nginx-55649fd747-xs7qg            1/1     Running   0          21s
virt-launcher-ubuntu-test-sdpk6   1/1     Running   0          24m
```

- [x]  Backup to external S3 storage 
![image](https://user-images.githubusercontent.com/29251855/144203185-ac06b59e-d617-4b24-b04f-028bb8b6af13.png)

### Test Information
* Test Environment: 3 nodes harvester on local kvm machines (vagrant ipxe example)
* Harvester version: master-1e042e04-head (12/01)


### Verify Steps

1. Clone the ipxe example in the offline feature branch
2. Create 3 nodes harvester by using vagrant pxe example
3. Open settings
4. Provide the following value in the http-proxy settings
```
{
  "httpProxy": "http://192.168.0.254:3128",
  "httpsProxy": "http://192.168.0.254:3128",
  "noProxy": "localhost,127.0.0.1,0.0.0.0,10.0.0.0/8,cattle-system.svc,192.168.0.0/16,.svc,.cluster.local,example.com"
}
```
5. Create image from URL 
https://cloud-images.ubuntu.com/focal/20211116/focal-server-cloudimg-amd64.img
6. ssh to harvester server node
7. Run `kubectl create deployment nginx --image=nginx:latest` on Harvester cluster
8. Run `kubectl get pods` 
9. Create a virtual machine
10. Set S3 back target setting
```
{
  "type": "s3",
  "endpoint": "",
  "accessKeyId": "",
  "secretAccessKey": "",
  "bucketName": "davidtclin-test",
  "bucketRegion": "us-east-1",
  "cert": "",
  "virtualHostedStyle": false
}
```
11. Take backup from virtual machine
